### ISSUE TITLE
Locate the part of the project to be updated

### ISSUE THREAD

**Issue by ofanobilbao** [ebi-ait/dcp-ingest-central#76]
Wranglers /contributors will know the “target” they wish to update, but it must not be assumed that they will know the UUID of this entity. An intuitive and user-friendly mechanism for retrieval of target entities is therefore required (e.g. search by ID/name/accession within a submission).

**Comment by ESapenaVentura** [comment]
Overall I like the list included in the acceptance criteria.

The first part is alright for the examples, but the `metadata terms` part I am not super convinced of. I do see some potential in people looking for `specimen from organism` just searching for all the specimens, but I don't see anyone filling in full qualified keys to find anything (especially since you have to provide a value for the search to make sense, e.g. `biomaterial_core.biomaterial_id`="blablabla")

I have made a recompilation of the list of fields that I would find useful to do searches on:

- `x.biomaterial_core.biomaterial_id`
- `x.biomaterial_core.biomaterial_name`
- `x.biomaterial_core.biosamples_accession` (Sometimes I might want to search by accessions)
- `x.biomaterial_core.insdc_sample_accession`
- `y.file_core.file_name`
- `process.process_core.process_id` (Useful for library preps)
- `z.protocol.protocol_core.protocol_id`
- `z.protocol.protocol_core.protocol_name`
- `process.insdc_experiment.insdc_experiment_accession`
- `sequence_file.insdc_run_accessions.[]`
- `project.contributors.[].name`
- `project.contributors.[].project_role.label`

Substituting:
- `x` by any biomaterial type
- `y` by any file type
- `z` by any protocol type

And also a list of fields that would also be nice to look for:
- `donor_organism.diseases.[].label` (or text)
- `donor_organism.development_stage.label` (or text)
- `specimen_from_organism.organ.label`
- `specimen_from_organism.organ_parts.[].label`
- `cell_suspension.selected_cell_types.[].label`

So, tl;dr, this list applies to the first example in the terms on the acceptance criteria and, about the second, I would find useful to look for entity types, but not for specific fields

**Comment by aaclan-ebi** [comment]
Tasks:
1. (new ticket?) [Core] Search endpoint
Do we need 1 endpoint? or should we create multiple requests to multiple endpoints?
2. (new ticket) [UI] Implement UI changes according to mockup
Search box and filtering the UI list based on the search

**Comment by gabsie** [comment]
hi everyone
Issue 106 - is where I have placed some suggestions for discussion - for search within a project

**Comment by ofanobilbao** [comment]
@gabsie I see this as related to the Updates Quicker. Do we want to keep it or recreate the info we already have here? Things might have changed over the last year. Thoughts?
